users/thystips/thystips.c

```diff
@@ -0,0 +1,240 @@
+        if (timer_elapsed(timeout_timer) >= 60000) { // 1 minute tick
+            timeout_counter++;
+            timeout_timer = timer_read();
+        }
+#    ifdef RGB_MATRIX_ENABLE
+        if (timeout_threshold > 0 && timeout_counter >= timeout_threshold) {
+            rgb_matrix_disable_noeeprom();
+        }
+#    endif
+    } // timeout_threshold = 0 will disable timeout
+}
+
+#endif // IDLE_TIMEOUT_ENABLE
+
+#if defined(ALTTAB_SCROLL_ENABLE) || defined(IDLE_TIMEOUT_ENABLE) // timer features
+__attribute__((weak)) void matrix_scan_keymap(void) {}
+
+void matrix_scan_user(void) {
+#    ifdef ALTTAB_SCROLL_ENABLE
+    encoder_tick_alttabscroll();
+#    endif
+#    ifdef IDLE_TIMEOUT_ENABLE
+    timeout_tick_timer();
+#    endif
+    matrix_scan_keymap();
+}
+#endif // ALTTAB_SCROLL_ENABLE or IDLE_TIMEOUT_ENABLE
+
+// PROCESS KEY CODES
+__attribute__((weak)) bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_keymap(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+        case KC_00:
+            if (record->event.pressed) {
+                // when keycode KC_00 is pressed
+                SEND_STRING("00");
+            }
+            break;
+        case KC_WINLCK:
+            if (record->event.pressed) {
+                keymap_config.no_gui = !keymap_config.no_gui; // toggle status
+            }
+            break;
+
+#ifdef IDLE_TIMEOUT_ENABLE
+        case RGB_TOI:
+            if (record->event.pressed) {
+                timeout_update_threshold(true);
+            }
+            break;
+        case RGB_TOD:
+            if (record->event.pressed) {
+                timeout_update_threshold(false); // decrease timeout
+            }
+            break;
+#endif // IDLE_TIMEOUT_ENABLE
+#ifdef RGB_MATRIX_ENABLE
+        case RGB_NITE:
+            if (record->event.pressed) {
+                rgb_nightmode = !rgb_nightmode;
+            }
+            break;
+        case RGB_DEF:
+            if (record->event.pressed) {
+                set_default_rgb();
+            }
+            break;
+#endif // RGB_MATRIX_ENABLE
+
+#ifdef EMOTICON_ENABLE
+        case EMO_SHRUG:
+            if (record->event.pressed) SEND_STRING("`\\_(''/)_/`");
+            break;
+        case EMO_CONFUSE:
+            if (record->event.pressed) SEND_STRING("(*_*)");
+            break;
+        case EMO_TEARS:
+            if (record->event.pressed) SEND_STRING("(T_T)");
+            break;
+        case EMO_NERVOUS:
+            if (record->event.pressed) SEND_STRING("(~_~;)");
+            break;
+        case EMO_JOY:
+            if (record->event.pressed) SEND_STRING("(^o^)");
+            break;
+        case EMO_SAD:
+            if (record->event.pressed) SEND_STRING(":'-(");
+            break;
+        case EMO_TFLIP:
+            if (record->event.pressed) SEND_STRING("(ノಠ痊ಠ)ノ彡┻━┻");
```

Comment:
This won't work, as it's attempting to send unicode, which isn't supported.

Reply:
This doesn't work very well on linux and I don't use it.